src/main/java/jenkins/plugins/openstack/compute/OpenStackDSL.java

```diff
@@ -0,0 +1,26 @@
+package jenkins.plugins.openstack.compute;
+
+import groovy.lang.Binding;
+import hudson.Extension;
+import org.jenkinsci.plugins.workflow.cps.CpsScript;
+import org.jenkinsci.plugins.workflow.cps.GlobalVariable;
+
+@Extension public class OpenStackDSL extends GlobalVariable {
+
+    @Override public String getName() {
+        return "openstack";
+    }
+
+    @Override public Object getValue(CpsScript script) throws Exception {
+        Binding binding = script.getBinding();
+        Object openstack;
+        if (binding.hasVariable(getName())) {
+            openstack = binding.getVariable(getName());
+        } else {
+            openstack = script.getClass().getClassLoader().loadClass("jenkins.plugins.openstack.compute.OpenStack").getConstructor(CpsScript.class).newInstance(script);
```

Comment:
jenkins.plugins.openstack.compute.Openstack is attached as a resource, not as a compiled class. Its loaded as a user written Groovy scripts, applying the same limitations as such (must be serializable). Ikts following the same structure as the [equivalent docker plugin](https://github.com/jenkinsci/docker-workflow-plugin/blob/master/src/main/resources/org/jenkinsci/plugins/docker/workflow/Docker.groovy)